tests/Unit/src/handlers/LiDARHandler.cc

```diff
@@ -0,0 +1,294 @@
+ * @brief Test fixture for LiDARHandler class.
+ *
+ *
+ * @author Eli Byrd (edbgkk@mst.edu)
+ * @date 2025-05-20
+ ******************************************************************************/
+class LiDARHandlerTests : public TestingBase<LiDARHandlerTests>
+{
+    protected:
+        std::string m_szDbPath = "../data/LiDAR/data/databases/Fugitive.db";
+
+    public:
+        LiDARHandlerTests()           = default;
+        ~LiDARHandlerTests() override = default;
+
+        void TestSetup() override {}
+
+        void TestTeardown() override {}
+};
+
+/******************************************************************************
+ * @brief Confirm that handler opens the database correctly.
+ *
+ *
+ * @author Eli Byrd (edbgkk@mst.edu)
+ * @date 2025-05-20
+ ******************************************************************************/
+TEST_F(LiDARHandlerTests, CanInitializeWithValidDB)
+{
+    LiDARHandler handler;
+    EXPECT_TRUE(handler.OpenDB(m_szDbPath));
+}
+
+/******************************************************************************
+ * @brief Confirm that initialization fails with nonexistent DB.
+ *
+ * @author Eli Byrd (edbgkk@mst.edu)
+ * @date 2025-05-20
+ ******************************************************************************/
+TEST_F(LiDARHandlerTests, FailsToInitializeInvalidDB)
+{
+    LiDARHandler handler;
+    EXPECT_FALSE(handler.OpenDB("invalid/path/to.db"));
+}
+
+/******************************************************************************
+ * @brief Confirm that handler can close the database.
+ *
+ *
+ * @author clayjay3 (claytonraycowen@gmail.com)
+ * @date 2025-07-14
+ ******************************************************************************/
+TEST_F(LiDARHandlerTests, CanCloseDB)
+{
+    LiDARHandler handler;
+    EXPECT_TRUE(handler.OpenDB(m_szDbPath));
+    EXPECT_TRUE(handler.CloseDB());
+    EXPECT_FALSE(handler.IsDBOpen());
+}
+
+/******************************************************************************
+ * @brief Confirm that closing without opening is safe.
+ *
+ *
+ * @author clayjay3 (claytonraycowen@gmail.com)
+ * @date 2025-07-14
+ ******************************************************************************/
+TEST_F(LiDARHandlerTests, CloseDBWithoutOpenIsSafe)
+{
+    LiDARHandler handler;
+    EXPECT_TRUE(handler.CloseDB());
+    EXPECT_FALSE(handler.IsDBOpen());
+}
+
+/******************************************************************************
+ * @brief Confirm that double open closes previous and opens new DB.
+ *
+ *
+ * @author clayjay3 (claytonraycowen@gmail.com)
+ * @date 2025-07-14
+ ******************************************************************************/
+TEST_F(LiDARHandlerTests, DoubleOpenClosesPrevious)
+{
+    LiDARHandler handler;
+    EXPECT_TRUE(handler.OpenDB(m_szDbPath));
+    // Open again with same path, should close and reopen
+    EXPECT_TRUE(handler.OpenDB(m_szDbPath));
+    EXPECT_TRUE(handler.IsDBOpen());
```

Comment:
Update to using the naming convention for variables.

```suggestion
    LiDARHandler stHandler;
    EXPECT_TRUE(stHandler.OpenDB(m_szDbPath));
    // Open again with same path, should close and reopen
    EXPECT_TRUE(stHandler.OpenDB(m_szDbPath));
    EXPECT_TRUE(stHandler.IsDBOpen());
```